### ISSUE TITLE
Required property DataSource.token leads to API client deserialization issues

### ISSUE THREAD

**Issue by trietsch** [opendatadiscovery/odd-platform#1456]
**Describe the bug**
After using the [OpenAPI Spec](https://github.com/opendatadiscovery/odd-platform/blob/17afca9b4f40237cb15908dfbe510620f34d45f4/odd-platform-specification/openapi.yaml) to generate an OpenAPI Kotlin client to communicate with a freshly deployed ODD Platform instance (with the samples added by the injector), and using the `SearchApi` to find all Datasets, I ran into the issue that the JSON cannot be deserialized into the generated Kotlin data class.

Example JSON:
```json
{
  "items": [
    {
      "id": 111,
      "oddrn": "//snowflake/host/abcxyz.eu-central-1.snowflakecomputing.com/databases/COMPUTE_WH/databases/SNOWFLAKE_SAMPLE_DATA/schemas/TPCDS_SF100TCL/tables/CATALOG_RETURNS",
      "external_name": "CATALOG_RETURNS",
      "internal_name": null,
      "ownership": null,
      "data_source": {
        "id": 9,
        "namespace": {
          "id": 5,
          "name": "Samples"
        },
        "token": null,
        "oddrn": "//snowflake/host/abcxyz.eu-central-1.snowflakecomputing.com/databases/COMPUTE_WH",
        "name": "Snowflake Sample Data",
        "description": "Snowflake sample data from the free trial"
      },
      "entity_classes": [...],
      "type": {
        "id": 1,
        "name": "TABLE"
      },
      "data_entity_groups": [],
      "status": {
        "status": "UNASSIGNED",
        "status_switch_time": null
      },
      "source_created_at": "2021-11-10T05:04:44.587Z",
      "source_updated_at": "2021-11-10T06:12:29.742Z",
      "last_ingested_at": "2023-10-03T12:53:06.964546Z",
      "view_count": 6,
      "is_stale": false,
      "stats": {
        "rows_count": 14405363575,
        "fields_count": 27,
        "consumers_count": 0
      },
      "source_list": null,
      "target_list": null,
      "input_list": null,
      "output_list": null,
      "suite_name": null,
      "suite_url": null,
      "expectation": null,
      "severity": null,
      "datasets_list": null,
      "linked_url_list": null,
      "latest_run": null,
      "items_count": null,
      "entities": null
    },
    ...
  ],
  "page_info": {
    "total": 466,
    "hasNext": true
  }
}
```

Note that `data_source.token` is null, which according the the [OAS is not possible](https://github.com/opendatadiscovery/odd-platform/blob/17afca9b4f40237cb15908dfbe510620f34d45f4/odd-platform-specification/components.yaml#L1196). As the Kotlin data class is generated from the OAS, it looks as follows:
```kotlin
data class DataSource (
    @field:JsonProperty("id")
    val id: kotlin.Long,

    @field:JsonProperty("token")
    val token: Token,

    @field:JsonProperty("oddrn")
    val oddrn: kotlin.String,

    @field:JsonProperty("name")
    val name: kotlin.String,

    @field:JsonProperty("namespace")
    val namespace: Namespace? = null,

    @field:JsonProperty("description")
    val description: kotlin.String? = null
)
```

Note that `token` is not nullable. Trying to deserialize the JSON into an instance of this data class results in:
```
Exception in thread "main" com.fasterxml.jackson.module.kotlin.MissingKotlinParameterException: Instantiation of [simple type, class org.opendatadiscovery.generated.model.DataSource] value failed for JSON property token due to missing (therefore NULL) value for creator parameter token which is a non-nullable type
 at [Source: (StringReader); line: 1, column: 514] (through reference chain: org.opendatadiscovery.generated.model.DataEntityList["items"]->java.util.ArrayList[0]->org.opendatadiscovery.generated.model.DataEntity["data_source"]->org.opendatadiscovery.generated.model.DataSource["token"])
	at com.fasterxml.jackson.module.kotlin.KotlinValueInstantiator.createFromObjectWith(KotlinValueInstantiator.kt:84)
        ...
```

**Set up**
Default docker-compose setup, as explained in the [docs](https://docs.opendatadiscovery.org/configuration-and-deployment/trylocally).

**Steps to Reproduce**

1. Follow the `try-locally` setup
2. Ensure that the dummy data is loaded
3. Try to deserialize a search result in Kotlin (if necessary, I can provide a MWE)

**Expected behavior**
The deserialization is performed correctly. Suggested change: make `token` nullable / not required.

**Comment by trietsch** [comment]
Found another one: `DataSetFieldStat.name` [shouldn't be null](https://github.com/opendatadiscovery/odd-platform/blob/17afca9b4f40237cb15908dfbe510620f34d45f4/odd-platform-specification/components.yaml#L1453) but it actually is in the DataSet API (get dataset structure latest).

**Comment by AndreyNenashev** [comment]
Hello, @trietsch! Yes, you are right. This is the collision between specification and the response. We will add this to our roadmap(will try to fix it before the next release), thanks for pointing out!

**Comment by RamanDamayeu** [comment]
closed by #1466
